### PR TITLE
Buffs cades against projectiles

### PR DESCRIPTION
**Pull request by Zonespace27**
# About the pull request

Barricades and tables are now better against projectiles.
Assuming a standard rifleman with an unmodified m41a vs. a barricade, the probabilities of hitting are:

| Dist | Old | New | New (Xeno)
|---|---|---|---|
|  1 |  0 | 0   | 0|
|  2 |  -0.1833 |  0.0045 |0|
|  3 | -0.0466  |  0.52 |0|
|  4 |  0.0550 |  0.85 |0|
|  5 |  0.2367  |  0.85 |0.0045|
|  [6, ∞) | 0.3383  | 0.85  |0.52|

# Explain why it's good for the game

Cades are completely worthless in hvh situations, because they fail to block the overwhelming majority of bullets (especially in cqc, where most gunfights happen). This makes them better while still allowing people who get within 2 tiles to attack easier.

# Changelog
:cl:
balance: Barricades are now far better at blocking bullets from the front. They will not block most bullets if the shooter is within 2 tiles, however.
/:cl:

### CODE REVIEW

**Comment by Git-Nivrak** [comment]
Bullets being blocked from 2 and 3 tiles is *very* very bad considering an FOB defense situation
You are basically forcing a lot of weapons to be adjacent to cades now (not talking about hvh)

Perhaps make it so this new chance is only applied for bullets coming from the front side of the cade?

**Comment by ihatethisengine** [comment]
Probably should have higher change to block bullets only from the front?

**Comment by Zonespace27** [comment]
> Bullets being blocked from 2 and 3 tiles is _very_ very bad considering an FOB defense situation You are basically forcing a lot of weapons to be adjacent to cades now (not talking about hvh)
> 
> Perhaps make it so this new chance is only applied for bullets coming from the front side of the cade?

Rectified

> Probably should have higher change to block bullets only from the front?

It now does

**Comment by RainbowStalin** [comment]
Don't know if this should be another PR, but i still think the Anti-FF cade and sandbags in particular could be buffed in terms of bullet resistance. Because you can still very easily shoot them down with a single mag (UPP that is).

**Comment by Zonespace27** [comment]
> Don't know if this should be another PR, but i still think the Anti-FF cade and sandbags in particular could be buffed in terms of bullet resistance. Because you can still very easily shoot them down with a single mag (UPP that is).

Out of scope for this PR

**Comment by Git-Nivrak** [comment]
Does this also affect xeno projectiles? Spits etc? If it does it probably shouldn't

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -668,10 +668,9 @@
 
 	//an object's "projectile_coverage" var indicates the maximum probability of blocking a projectile
 	var/effective_accuracy = P.get_effective_accuracy()
-	var/distance_limit = 6 //number of tiles needed to max out block probability
 	var/accuracy_factor = 50 //degree to which accuracy affects probability   (if accuracy is 100, probability is unaffected. Lower accuracies will increase block chance)
 
-	var/hitchance = min(projectile_coverage, (projectile_coverage * distance/distance_limit) + accuracy_factor * (1 - effective_accuracy/100))
+	var/hitchance = min(projectile_coverage, (projectile_coverage * distance / (projectile_coverage_distance_limit * (cade_direction_correct ? 3 : 1))) + accuracy_factor * (1 - effective_accuracy/100))
```
Comment (Drulikar) [inline code/modules/projectiles/projectile.dm:673]: xeno projectiles are affected by this change. At a minimum they should have a a similar mitigation like hardpoints do.

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile.dm (hunk 1 above)
Comment (Zonespace27) [reply code/modules/projectiles/projectile.dm:673]: Do those numbers (see PR desc) look good?

Comment (Drulikar) [reply code/modules/projectiles/projectile.dm:673]: They seem fine as long as that's not the numbers for xenos.

**Comment by Drulikar** [comment]
Undraft when you're happy with it.

**Review by Drulikar — APPROVED** [review]

**Pull request merged**